contracts/WSTETHPriceFeed.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.15;
+
+import "./vendor/@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
+
+interface IWstETH {
+    function decimals() external view returns (uint8);
+    function tokensPerStEth() external view returns (uint256);
+}
+
+contract WstETHPriceFeed is AggregatorV3Interface {
+    /** Custom errors **/
+    error InvalidInt256();
+    error NotImplemented();
+
+    string public constant override description = "Custom price feed for wstETH / USD";
+
+    uint public constant override version = 1;
+
+    /// @notice Scale for returned prices
+    uint8 public override decimals = 8;
+
+    /// @notice Chainlink stETH / USD price feed
+    address public immutable stETHtoUSDPriceFeed;
+
+    /// @notice WstETH contract address
+    address public immutable wstETH;
+
+    constructor(address stETHtoUSDPriceFeed_, address wstETH_) {
+        stETHtoUSDPriceFeed = stETHtoUSDPriceFeed_;
+        wstETH = wstETH_;
+    }
+
+    function signed256(uint256 n) internal pure returns (int256) {
+        if (n > uint256(type(int256).max)) revert InvalidInt256();
+        return int256(n);
+    }
+
+    /**
+     * @notice Unimplemented function required to fulfill AggregatorV3Interface; always reverts
+     **/
+    function getRoundData(uint80 _roundId) override external view returns (
+        uint80 roundId,
+        int256 answer,
+        uint256 startedAt,
+        uint256 updatedAt,
+        uint80 answeredInRound
+    ) {
+        revert NotImplemented();
```

Comment:
`getRoundData` is required for the AggregatorV3Interface, but it seems potentially harmful to return a value for this function since we don't actually know what the `tokensPerStEth` value was at the time that of a given `stETH / USD` price